src/pip/_internal/commands/install.py

```diff
@@ -341,6 +352,17 @@ def run(self, options: Values, args: List[str]) -> int:
                 reqs, check_supported_wheels=not options.target_dir
             )
 
+            if options.dry_run:
+                items = [
+                    f"{item.name}-{item.metadata['version']}"
```

Comment:
what do you think using == or spaces instead of "-"? (related https://github.com/pypa/pip/issues/10997)

Reply:
I think we should be consistent with what a regular install prints. We can change both later if and when #10997 concludes a change is needed.